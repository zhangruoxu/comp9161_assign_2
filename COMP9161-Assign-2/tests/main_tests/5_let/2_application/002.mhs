main
     = 
    let g =
        letfun g' x = x * x;

    in let 
        f = g;  -- :: Int -> Int
    
    in g 7;
