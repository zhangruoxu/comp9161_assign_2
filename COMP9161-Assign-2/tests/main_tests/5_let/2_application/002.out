main :: Int = let g :: (Int -> Int) = (letfun g' :: (Int -> Int) x = (*) x x);
              in let f :: (Int -> Int) = g;
                 in g 7;
